lib/matplotlib/tests/test_axes.py

```diff
@@ -5782,3 +5782,46 @@ def test_zoom_inset():
                    [0.8425, 0.907692]])
     np.testing.assert_allclose(axin1.get_position().get_points(),
             xx, rtol=1e-4)
+
+
+def test_inset_codes():
+    "Test that the inset codes put the inset where we want..."
+    fig, ax = plt.subplots()
+    poss = [[[0.415625, 0.686111],
+             [0.609375, 0.886111]],
+            [[0.695833, 0.686111],
+             [0.889583, 0.886111]],
+            [[0.695833, 0.4     ],
+             [0.889583, 0.6     ]],
+            [[0.695833, 0.113889],
+             [0.889583, 0.313889]],
+            [[0.415625, 0.113889],
+             [0.609375, 0.313889]],
+            [[0.135417, 0.113889],
+             [0.329167, 0.313889]],
+            [[0.135417, 0.4     ],
+             [0.329167, 0.6     ]],
+            [[0.135417, 0.686111],
+             [0.329167, 0.886111]]]
+    codes = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+    for pos, code in zip(poss, codes):
+        axin1 = ax.inset_axes(code)
+        np.testing.assert_allclose(axin1.get_position().get_points(),
```

Comment:
Would it make sense to define some `bbox`es in axes coordinates here and compare to their coordinates instead? (Seems like this test is pretty susceptible to unrelated changes in the subplot positioning.)

Reply:
The string-code positioning has a borderpad that is in pixels, not axes co-ordinates, so knowing the bbox a-priori would just mean I was duplicating the code in the method.  I guess that'd test if someone changed it ...